### PR TITLE
Update tv5mondeplus.py to detect subtitles

### LINKED ISSUE #4205: TV5MondePlus: Download subtitles
### Checklist

- [X] I'm requesting a site-specific feature
- [X] I've verified that I'm running yt-dlp version **2022.06.22.1** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Europe

### Example URLs

https://revoir.tv5monde.com/toutes-les-videos/series-fictions/l-art-du-crime-un-coeur-de-pierre

### Description

When playing the stream in the browser, I can request subtitles.
However yt-dlp does not detect them.

image

### Verbose log

```shell
[debug] Command-line config: ['-vU', '--all-subs', 'https://revoir.tv5monde.com/toutes-les-videos/series-fictions/l-art-du-crime-un-coeur-de-pierre']
[debug] User config "/home/frank/.config/yt-dlp/config": ['--all-subs', '--convert-subs=srt', '-k', '--write-annotations', '--write-info-json', '--write-description', '--compat-options', 'filename']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.06.22.1 [a86e01e]
[debug] Compatibility options: filename
[debug] Python version 3.10.4 (CPython 64bit) - Linux-5.15.0-39-generic-x86_64-with-glibc2.35
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.11.0, brotli-1.0.9, certifi-2020.06.20, mutagen-1.45.1, pyxattr-0.7.2, secretstorage-3.3.1, sqlite3-2.6.0, websockets-9.1
[debug] Proxy map: {}
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.06.22.1, Current version: 2022.06.22.1
yt-dlp is up to date (2022.06.22.1)
[debug] [TV5MondePlus] Extracting URL: https://revoir.tv5monde.com/toutes-les-videos/series-fictions/l-art-du-crime-un-coeur-de-pierre
[TV5MondePlus] l-art-du-crime-un-coeur-de-pierre: Downloading webpage
[TV5MondePlus] l-art-du-crime-un-coeur-de-pierre: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 0d1de55b-40a2-5635-c2a2-77c4c5e225b1: Downloading 1 format(s): hls-2485+hls-259
[info] Writing video description to: Un coeur de pierre S5EP1_2-0d1de55b-40a2-5635-c2a2-77c4c5e225b1.description
[info] Writing video metadata as JSON to: Un coeur de pierre S5EP1_2-0d1de55b-40a2-5635-c2a2-77c4c5e225b1.info.json
WARNING: There are no annotations to write.
[SubtitlesConvertor] There aren't any subtitles to convert
[debug] Invoking hlsnative downloader on "https://msod.tv5monde.com/tv5monde/vod/assets/106982361_6D4BA7b/materials/N8JxIH05Nk_6D4BA7b/vod-idx.ism/vod-idx-video=2100000.m3u8"
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 734

etc.
at the end, no subtitles are present.

yt-dlp -J does not list subtitles:
"requested_subtitles": null,
```

### PR DESCRIPTION
**Pull request by FrankZ85**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

---

### Description of your *pull request* and other information
Fixes #4205

This retrieves the subtitles as announced by the streaming site. The code to iterate over the subtitle definitions is taken from other extractors.

It would still be nice to try and detect the other, not announced, subtitles. Maybe by sending HEAD requests to the guessed URLs to get the HTTP status? For this entry, replacing the second "FRA" by "DEU", "ENG", "SPA" worked just fine.

### CODE REVIEW

File: yt_dlp/extractor/tv5mondeplus.py [hunk 1]
```diff
@@ -48,6 +49,18 @@ class TV5MondePlusIE(InfoExtractor):
...
+    @staticmethod
+    def _extract_subtitles(data_captions):
+        subtitles = {}
+        for f in try_get(data_captions, lambda x: x['files'], list) or []:
+            subtitle_url = url_or_none(f.get('file'))
+            if subtitle_url:
+                lang = f.get('label', 'fra')
+                subtitles.setdefault(lang, []).append({
+                    'url': subtitle_url,
+                })
```
Comment (pukkandan) [inline yt_dlp/extractor/tv5mondeplus.py:61]:
```suggestion
                lang = f.get('label') or 'fra'
                subtitles.setdefault(lang, []).append({'url': subtitle_url})
```

File: yt_dlp/extractor/tv5mondeplus.py [hunk 2]
```diff
@@ -95,6 +108,12 @@ def _real_extract(self, url):
         if series and series != title:
             title = '%s - %s' % (series, title)
 
+        subtitles = {}
+        if 'data-captions' in vpl_data:
+            captions = self._parse_json(
+                vpl_data['data-captions'], display_id)
+            subtitles = self._extract_subtitles(captions)
```
Comment (pukkandan) [inline yt_dlp/extractor/tv5mondeplus.py:115]:
```suggestion
        subtitles = self._extract_subtitles(self._parse_json(
            vpl_data.get('data-captions', '{}'), video_id, fatal=False))
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

**Comment by pukkandan** [comment]
confirm evevrything's still working

File: yt_dlp/extractor/tv5mondeplus.py [hunk 3]
```diff
@@ -95,6 +106,9 @@ def _real_extract(self, url):
         if series and series != title:
             title = '%s - %s' % (series, title)
 
+        subtitles = self._extract_subtitles(self._parse_json(
+            vpl_data.get('data-captions', '{}'), video_id, fatal=False))
```
Comment (pukkandan) [inline yt_dlp/extractor/tv5mondeplus.py:110]:
```suggestion
            vpl_data.get('data-captions', '{}'), display_id, fatal=False))
```

**Comment by seproDev** [comment]
Can confirm still works

File: yt_dlp/extractor/tv5mondeplus.py [hunk 4]
```diff
@@ -178,4 +191,5 @@ def process_video_files(v):
             'formats': formats,
             'series': series,
             'episode': episode,
+            'subtitles': subtitles,
```
Comment (bashonly) [inline yt_dlp/extractor/tv5mondeplus.py:194]:
```suggestion
            'formats': formats,
            'subtitles': self._extract_subtitles(self._parse_json(
                traverse_obj(vpl_data, ('data-captions', {str}), default='{}'), display_id, fatal=False)),
            'series': series,
            'episode': episode,
```

File: yt_dlp/extractor/tv5mondeplus.py [hunk 5]
```diff
@@ -84,6 +84,16 @@ class TV5MondePlusIE(InfoExtractor):
...
+    @staticmethod
+    def _extract_subtitles(data_captions):
+        subtitles = {}
+        for f in try_get(data_captions, lambda x: x['files'], list) or []:
+            subtitle_url = url_or_none(f.get('file'))
+            if subtitle_url:
+                lang = f.get('label') or 'fra'
+                subtitles.setdefault(lang, []).append({'url': subtitle_url})
+        return subtitles
+
```
Comment (bashonly) [inline yt_dlp/extractor/tv5mondeplus.py:96]:
```suggestion
    @staticmethod
    def _extract_subtitles(data_captions):
        subtitles = {}
        for f in traverse_obj(data_captions, ('files', lambda _, v: url_or_none(v['file']))):
            subtitles.setdefault(f.get('label') or 'fra', []).append({'url': f['file']})
        return subtitles
```

**Review by bashonly — APPROVED** [review]

**Pull request merged**